koboldcpp: update `cudaArches` example, remove `gitUpdater`, drop `openblas`, 1.79.1 -> 1.80.1

4evy opened the pull request:
Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.81
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.79.1...v1.80.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

4evy committed:
koboldcpp: update `cudaArches` example

4evy committed:
koboldcpp: remove `gitUpdater`

4evy committed:
koboldcpp: drop `openblas`

Upstream dropped `openblass` since v1.75

4evy committed:
koboldcpp: 1.79.1 -> 1.80

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.80
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.79.1...v1.80

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366837 --extra-nixpkgs-config '{ allowAliases = false; allowBroken = false; allowUnfree = true; checkMeta = true; cudaSupport = true; cudaCapabilities = [ "7.5" ]; }'`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366837`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

4evy committed:
koboldcpp: add `(darwinMinVersionHook "10.15")`

Bump up minimum version, so no APIs are missing

4evy committed:
koboldcpp: disable vulkan for darwin

It's causing build failure

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366837`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

4evy committed:
koboldcpp: 1.80 -> 1.80.1

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.80.1
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.80...v1.80.1

paparodeo commented on pkgs/by-name/ko/koboldcpp/package.nix:
```diff
@@ -65,8 +62,10 @@ effectiveStdenv.mkDerivation (finalAttrs: {
-    ++ lib.optionals openblasSupport [ openblas ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_12
+      (darwinMinVersionHook "10.15")
```
this isn't needed on master anymore.. both x64 and aarch64 now default to 11.

DontEatOreo replied:
Thanks for letting me know. When I made this PR initially, this wasn't the case, those were docs:

https://github.com/NixOS/nixpkgs/blob/62501808a721f34d429f0bde2adb9ddcdda66f89/doc/stdenv/platform-notes.chapter.md?plain=1#L92-L103

paparodeo replied:
np. tho if you are planning on back porting this into release-24.11 it is still needed there as the x64 darwin sdk is still at 10.12 and minVersion is 10.12 as well.
